packages/apidom-ns-workflows-1/test/predicates.ts

```diff
@@ -287,4 +289,62 @@ describe('predicates', function () {
       assert.isFalse(isSourceDescriptionsElement(sourceDescriptionsElementSwan));
     });
   });
+
+  context('isCriterionElement', function () {
+    context('given CriterionElement instance value', function () {
+      specify('should return true', function () {
+        const element = new CriterionElement();
+
+        assert.isTrue(isCriterionElement(element));
+      });
+    });
+
+    context('given subtype instance value', function () {
+      specify('should return true', function () {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        class CriterionSubElement extends CriterionElement {}
+
+        assert.isTrue(isCriterionElement(new CriterionSubElement()));
+      });
+    });
+
+    context('given non CriterionSubElement instance value', function () {
+      specify('should return false', function () {
+        assert.isFalse(isCriterionElement(1));
+        assert.isFalse(isCriterionElement(null));
+        assert.isFalse(isCriterionElement(undefined));
+        assert.isFalse(isCriterionElement({}));
+        assert.isFalse(isCriterionElement([]));
+        assert.isFalse(isCriterionElement('string'));
+      });
+    });
+
+    specify('should support duck-typing', function () {
+      const CriterionElementDuck = {
+        _storedElement: 'criterion',
+        _content: [],
+        classes: new ArrayElement(['criterion']),
```

Comment:
Not in original `CriterionElement` definition.

Reply:
added to definition